### PR TITLE
New Images Widget

### PR DESCRIPTION
**Pull request by marcelpinto**
- Showcase how to use Coil + WM to load images
- Using Exact mode to resize images

### CODE REVIEW

File: AppWidget/app/src/main/java/com/example/android/appwidget/glance/image/ImageWorker.kt [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            .build()
+
+        // Request the image to be loaded and throw error if it failed
+        with(context.imageLoader) {
+            if (force) {
+                diskCache?.remove(url)
+                memoryCache?.remove(MemoryCache.Key(url))
+            }
+            val result = execute(request)
+            if (result is ErrorResult) {
```
Comment (yschimke) [inline AppWidget/app/src/main/java/com/example/android/appwidget/glance/image/ImageWorker.kt:147]: If you extracted this to a helper, then throwing on ErrorResult isn't probably the right option. ImageRequest supports error images, so maybe best to check if the response.drawable is non null.

**Review by yaraki — APPROVED** [review]

File: AppWidget/app/src/main/java/com/example/android/appwidget/glance/image/ImageWorker.kt (hunk 1 above)
Comment (marcelpinto) [reply AppWidget/app/src/main/java/com/example/android/appwidget/glance/image/ImageWorker.kt:147]: I would need to implement a more complex "state" for the widget with success/error images then. Instead here I decided to throw the error so the worker fails and retries using exponential backoff. 

But agree that probably could be extracted into a more reusable method. But for the sake of the sample I think it's enough

Comment (yschimke) [reply AppWidget/app/src/main/java/com/example/android/appwidget/glance/image/ImageWorker.kt:147]: Coil has placeholder images also, so a proper library might even use that depending on the delay fetching images.

**Comment by marcelpinto** [comment]
The problem is those might not be stored in disk. Thus, no URI available to
pass. Only the downloaded image is cached

**Pull request merged**